mode/soy/soy.js

```diff
@@ -517,14 +517,14 @@
             }
             return expression(stream, state);
           case "literal":
-            if (stream.match(/^(?=\{\/literal})/)) {
+            if (stream.match('{/literal}', false)) {
```

Comment:
Looking ahead via `(?=…)` does have the same effect as not consuming the match (therefor the `false`). The only benefit of this change is that it replaces a regex with a faster string match.